analysis/table_age/table_age.R

```diff
@@ -0,0 +1,205 @@
+# Load libraries ---------------------------------------------------------------
+print("Load libraries")
+
+library(magrittr)
+library(here)
+library(dplyr)
+library(jsonlite)
+library(skimr)
+
+# Define table_age output folder ---------------------------------------------------------
+print("Creating output/table_age output folder")
+
+table_age_dir <- "output/table_age/"
+fs::dir_create(here::here(table_age_dir))
+
+# Source common functions ------------------------------------------------------
+print("Source common functions")
+
+source("analysis/utility.R")
+
+# Load json file containing vax study dates ------------------------------------
+print('Load json file containing vax study dates')
+
+study_dates <- fromJSON("output/study_dates.json")
+
+# Specify arguments ------------------------------------------------------------
+print("Specify arguments")
+
+args <- commandArgs(trailingOnly = TRUE)
+
+print(length(args))
+
+if (length(args) == 0) {
+  cohort <- "vax"
+  age_str <- "40;45;55;60;65"
+  preex <- "All" # "All", TRUE, or FALSE
+} else {
+  cohort <- args[[1]]
+  age_str <- args[[2]]
+  if (length(args) < 3) {
+    preex <- "All"
+  } else {
+    preex <- args[[3]]
+  } # allow an empty input for the preex variable
+}
+
+age_bounds <- as.numeric(stringr::str_split(as.vector(age_str), ";")[[1]])
+
+# Load data --------------------------------------------------------------------
+print("Load data")
+
+df <- readr::read_rds(paste0(
+  "output/dataset_clean/input_",
+  cohort,
+  "_clean.rds"
+))
+
+# Table 1 Processing Start -----------------------------------------------------
+print("Table 1 processing")
+
+# Remove people with history of COVID-19 ---------------------------------------
+print("Remove people with history of COVID-19")
+
+df <- df[df$sub_bin_covidhistory == FALSE, ]
+
+# Create exposure indicator ----------------------------------------------------
+print("Create exposure indicator")
+
+df$exposed <- !is.na(df$exp_date_covid) # Do I add extra restrictions here to be within cohort limits?
```

Comment:
It doesn't really matter for the purposes of this table. We are  looking at the pattern for age of dementia diagnosis, not the actual number of diagnoses.